editor/src/clj/editor/game_object.clj

```diff
@@ -140,6 +140,13 @@
       (update :display-order (fn [display-order]
                                (vec (distinct (concat display-order (:display-order source-properties))))))))
 
+(defn resource-path-error [_node-id source-resource]
+  (let [node-type (some-> source-resource resource/resource-type :node-type)]
+    (or (validation/prop-error :fatal _node-id :path validation/prop-nil? source-resource "Path")
+        (validation/prop-error :fatal _node-id :path validation/prop-resource-not-exists? source-resource "Path")
+        (and (= node-type editor.code.script/ScriptNode)
```

Comment:
If a node is ScriptNode, we should make sure it it the only allowed `script` file here, nor `render_script` or `gui_script`